### PR TITLE
Add STRING input validator

### LINKED ISSUE #1122: Add a basic validation of flow inputs
### Feature description

It should be possible to specify at least a basic validation of flow input value. Such validation would prevent starting an execution using a clearly wrong input.

### PR DESCRIPTION
**Pull request by yuri1969**
This PR adds a regex-based validation for the `STRING` inputs. The pattern is defined by `input`'s newly added `validator` property.

``` yaml
inputs:
  - name: my-upper-case-value
    type: STRING
    validator: [A-Z]+
```

This PR deals with the STRING inputs only since these are frequent & variable.

Naming, etc. should be reviewed to fit in.

close #1122

### CODE REVIEW

**Comment by yuri1969** [comment]
@loicmathieu Great, adding type-specific validation implementations should be easy now.

**Comment by loicmathieu** [comment]
I rebased the branch onto master as there was a strange test error.

We will discuss internally if we merge this one as is or if we take the opportunity to add more constraints.

File: core/src/main/java/io/kestra/core/models/flows/Input.java [hunk 1]
```diff
@@ -35,6 +52,8 @@ public class Input {
 
     String defaults;
 
+    public abstract boolean isValid(String input);
```
Comment (tchiotludo) [inline core/src/main/java/io/kestra/core/models/flows/Input.java:55]:
```suggestion
  public abstract void validate(Object input) throws ConstraintViolationException {
```
- To allow a better error message on the ui
- The type can be object I think

File: core/src/test/java/io/kestra/core/serializers/YamlFlowParserTest.java [hunk 2]
```diff
@@ -112,18 +113,27 @@ void inputsFailed() {
         );
 
         assertThat(exception.getConstraintViolations().size(), is(2));
-        assertThat(exception.getConstraintViolations().stream().filter(r -> r.getPropertyPath().toString().equals("inputs[0].name")).findFirst().orElseThrow().getMessage(), containsString("must match"));
-        assertThat(exception.getConstraintViolations().stream().filter(r -> r.getPropertyPath().toString().equals("inputs[0].type")).findFirst().orElseThrow().getMessage(), is("must not be null"));
+        // FIXME the property path changes: io.kestra.core.models.flows.Flow["inputs"]->java.util.ArrayList[0]
```
Comment (tchiotludo) [inline core/src/test/java/io/kestra/core/serializers/YamlFlowParserTest.java:116]: Please explain?

**Review by tchiotludo — CHANGES_REQUESTED** [review]
Minor change for seems to be good stuff anyway

File: core/src/main/java/io/kestra/core/models/flows/Input.java (hunk 1 above)
Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/models/flows/Input.java:55]: Will do.
The type is always a string, the cast / parse to the real type of the object is done later. To change this I'll need more changes.

Comment (tchiotludo) [reply core/src/main/java/io/kestra/core/models/flows/Input.java:55]: I would prefer a proper type, if not, the cast will need to be double done for min or max for example, don't make sense

Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/models/flows/Input.java]: Done

**Review by tchiotludo — APPROVED** [review]

**Comment by loicmathieu** [comment]
@yuri1969 thanks a lot for your contribution, it is now merged and will be included in Kestra 0.9.0 !

I created two issue to continue on the subject to have a complete input validation in Kestra.

**Pull request merged**